java/src/com/ibm/streamsx/rest/Element.java

```diff
@@ -37,4 +42,31 @@ void setConnection(StreamsConnection connection) {
     public final String toString() {
         return pretty.toJson(this);
     }
+    
+    private Gson refreshJson;
+    
+    /**
+     * Refresh this element.
+     * 
+     * Attributes of this object are updated from current state
+     * though the Streams REST API.
+     * 
+     * @throws IOException Error communicating with Streams.
+     */
+    public void refresh() throws IOException {
+        String response = connection().getResponseString(self);
```

Comment:
it actually uses the URI from the "parent" object, but appends the name of the metric as a query parameter:

`?name=my_metric_name`

That though is returned as a list of one item, so we have to manually unpack the values that we expect to change.